### ISSUE TITLE
`llvm::BitTracker::UseQueueType::Uses` is constructed from uninitialized member

### ISSUE THREAD

**Issue by Eisenwave** [llvm/llvm-project#82319]
I got this warning when compiling LLVM with g++:
```
In file included from /home/user/projects/llvm-project/llvm/lib/Target/Hexagon/HexagonBitTracker.h:12,
                 from /home/user/projects/llvm-project/llvm/lib/Target/Hexagon/HexagonBitTracker.cpp:9:
/home/user/projects/llvm-project/llvm/lib/Target/Hexagon/BitTracker.h: In constructor ‘llvm::BitTracker::UseQueueType::UseQueueType()’:
/home/user/projects/llvm-project/llvm/lib/Target/Hexagon/BitTracker.h:75:27: warning: member ‘llvm::BitTracker::UseQueueType::Dist’ is used uninitialized [-Wuninitialized]
   75 |     UseQueueType() : Uses(Dist) {}
      |
```
This looks to be a true positive:
https://github.com/llvm/llvm-project/blob/119a72866f0e143127355fe6e03f57c4f8bab1ae/llvm/lib/Target/Hexagon/BitTracker.h#L74-L76

This performs initialization from an uninitialized member due to the order of declaration:
https://github.com/llvm/llvm-project/blob/119a72866f0e143127355fe6e03f57c4f8bab1ae/llvm/lib/Target/Hexagon/BitTracker.h#L103-L105
